extensions/amp-consent/0.1/tcf-api-command-manager.js

```diff
@@ -212,7 +209,6 @@ export class TcfApiCommandManager {
     const additionalData = {...sharedData, additionalConsent};
 
     return {
-      tcfPolicyVersion: TCF_POLICY_VERSION,
```

Comment:
Are we sure that removing `tcfPolicyVersion` will not break a protocol, maybe will be better to keep it and return 4 for TCF v2.2?

Reply:
Hi @sevriugin, If we want to keep it then we have to either extract it from the TCString and then add the amp-consent decoder here (which I think would then raise the question of what benefit there is in having amp-consent and not the CMP directly). Or give the responsibility to the CMP to tell which policy version it used. Since the tcf policy version is encoded inside the TCString, I thought this makes this information hardcoded into the amp-consent component useless and misleading. However, a vendor must decode the TCString to know if it is enabled or not.

Reply:
it could be solved with a solution that I specified in the issue. that is, when the consents from the cmp are applied, I add a field(tcf policy version) that will be saved via the storage management of amp and that data will then be taken up to be passed into the tcfapi return in this way by setting the tcf policy version to 2 by default if not present (if not specified by the cmp). behavior before this update will not change so let's say it's a backwards compatible solution. (I've heard that some are already using tcf v2.2 so in theory it's not that backwards compatible since the data would have to be migrated and at this point it would require the use of the tcf library directly inside amp, obviously for us who still have to switch it doesn't create any inconsistency, but they can update their data without causing re-prompt with the actual amp-consent implementation). in practice, the cmp, in addition to providing the data that we already provide, will also have to provide the tcf policy version, in this way future updates will be managed independently by the cmp. what do you think about it? (maybe it should also be optional as a parameter so as not to break what is currently running in production).
I want to know also what someone of amp project think about it before implementing it. Of course there is still the doubt about how we can say these things to all vendors, but this solution is the most retro-compatible (I think)

Reply:
I think we have an issue to support both TCF version for some period of time and also do not break the existing protocol. If we look at TCString and TCModel we don't have an issue as we getting `tcfPolicyVersion` from GVL, but for `__tcfapi` we having `tcfPolicyVersion` hardcoded. I don't know the amp codebase well but if we could manage to make `tcfPolicyVersion` depending on TCF version and having a way to set TCF version if need (default value could be 2 for now and 2.2 after 20th Nov) it will be great

Reply:
> I think we have an issue to support both TCF version for some period of time and also do not break the existing protocol. If we look at TCString and TCModel we don't have an issue as we getting `tcfPolicyVersion` from GVL, but for `__tcfapi` we having `tcfPolicyVersion` hardcoded. I don't know the amp codebase well but if we could manage to make `tcfPolicyVersion` depending on TCF version and having a way to set TCF version if need (default value could be 2 for now and 2.2 after 20th Nov) it will be great 

We could as I said, of course the CMP have to say the tcf policy version. This how works the solution I described, before. So when CMP use version 4 or any other version it will provide also the tcf policy version so the amp implementation will pass this to vendors. The tcf policy version if not specified by CMP will be 2 (this make the whole thing a backwards compatible).

Reply:
@powerivq I ask to you if you can help us in this update, and also what you think about the solution we discussed (if not, maybe you can assign to this someone can help us, since the deadline for CMPs to use the new tcf policy version 4 aka tcf v2.2 is the 20th of November). Of course we can do the same to ping everyone to know at least about the changes and how they can update their implementation.

Reply:
hey all trying to understand the changes so we can get this through ASAP. apologies for any dumb questions on my part as I am not familiar with amp-consent very well or TCF but I am currently going through documentation.

If im understanding correctly @marco-prontera (and correct me if i am wrong) , the proposal is for the CMP to declare the GVL that they are using (?) get that into the AMP system and use tcf 2.2 if the version is 4? else fallback to tcf v2.2?

Reply:
> If im understanding correctly @marco-prontera (and correct me if i am wrong) , the proposal is for the CMP to declare the GVL that they are using (?) get that into the AMP system and use tcf 2.2 if the version is 4? else fallback to tcf v2.2?

The proposal is for the CMP to declare the tcf policy version (related to the GVL they are using).
to preserve the current behavior and as a fallback the tcf policy version will be 2. (As it's used in tcf v2.2). This will guarantee a free deploy of that thing in amp.

I will update ASAP the PR to match to this proposal, its right for you?

Reply:
this sounds good to me. are there any drawbacks to the solution?

i feel like the proposal is good because the fallback seems reasonable (the current version that we previously have hardcoded), and the version seems to be explicit and needs to be declared by the CMP. 

@sevriugin seems to have indicated as well that he is OK with the proposal with the thumbs up

/cc @powerivq for any other input and for visibility

Reply:
LGTM

Reply:
@erwinmombay I have implemented changes to reflect the proposal. Tell me if I need to change anything

Reply:
thanks @marco-prontera ill start reviewing

Reply:
@powerivq @erwinmombay I think I forgot the part where the iframe through the postMessage can indicate the tcfPolicyVersion, I implement it immediately and update the documentation

Reply:
thanks for being thorough! and the documentation changes